### PR TITLE
Improve documentation of IndexMap

### LINKED ISSUE #1527: Should IndexMap be in MOI
`copy_to` doesn't define what type of index map should be returned. 

https://github.com/jump-dev/MathOptInterface.jl/blob/9e79136ce46e288cf0eeb4cc7fabeb895cb09cb2/src/MathOptInterface.jl#L126-L160

Should we let implementations return different types? Or should we standardize on `Utilities.IndexMap` and move it to MOI proper? 

I would vote moving to MOI proper, since this is a pretty key piece of the API, and it's effectively just this file: https://github.com/jump-dev/MathOptInterface.jl/blob/master/src/Utilities/copy/index_map.jl

### PR DESCRIPTION
**Pull request by odow**
Closes #1527 

Just a look at what this would look like.

### CODE REVIEW

**Comment by odow** [comment]
I guess if I'd tried this locally I would have found the problem.

We have a bootstrap issue: IndexMap needs Utilities.DoubleDicts,  but Utilities needs IndexMap. this was why it was in MOI.Utilities in the first place.

I may just add a `const IndexMap = Utilities.IndexMap` to the bottom of MOI.

File: src/MathOptInterface.jl [hunk 1]
```diff
@@ -124,23 +124,41 @@ This is used in two places to determine whether to add a cache:
...
-then an [`UnsupportedAttribute`](@ref) error is thrown. Unsupported *optimizer*
-attributes are treated differently:
-
-* If `warn_attributes` is `true`, a warning is displayed, otherwise,
-* the attribute is silently ignored.
+    copy_to(
+        dest::ModelLike,
+        src::ModelLike;
+        copy_names::Bool = true,
+        warn_attributes::Bool = true,
```
Comment (odow) [inline src/MathOptInterface.jl:131]: @blegat do we really use this `warn_attributes` thing? Perhaps we should just remove? We don't really copy optimizer attributes.

File: src/MathOptInterface.jl [hunk 2]
```diff
@@ -124,23 +124,41 @@ This is used in two places to determine whether to add a cache:
...
-or constraint attribute that is copied from `src` is not supported by `dest`
-then an [`UnsupportedAttribute`](@ref) error is thrown. Unsupported *optimizer*
-attributes are treated differently:
-
-* If `warn_attributes` is `true`, a warning is displayed, otherwise,
-* the attribute is silently ignored.
+    copy_to(
+        dest::ModelLike,
+        src::ModelLike;
+        copy_names::Bool = true,
```
Comment (odow) [inline src/MathOptInterface.jl:130]: Remind me why we need `copy_names`? Why can't we just try by default, and skip if they aren't supported?

File: src/MathOptInterface.jl (hunk 1 above)
Comment (blegat) [reply src/MathOptInterface.jl:131]: No we should remove in MOI v0.10! It's a good thing we didn't forget.

**Review by blegat — APPROVED** [review]
Let's address warn_attributes and copy_names in separate issues/PRs

**Pull request merged**